src/software/parameter/parameter.h

```diff
@@ -103,32 +68,24 @@ class Parameter
      *
      * @param callback The function to call when this parameter's value is changed
      */
-    void registerCallbackFunction(std::function<void(T)> callback)
+    void registerCallbackFunction(std::function<void(T)> callback) const
     {
         std::scoped_lock callback_lock(this->callback_mutex_);
         callback_functions.emplace_back(callback);
     }
 
-   private:
-    // this mutex is marked as "mutable" so that it can be acquired in a const function
+   protected:
+    // mutexes are marked as mutable so that they can be acquired in a const function
     mutable std::mutex value_mutex_;
-    std::mutex callback_mutex_;
+    mutable std::mutex callback_mutex_;
 
     // Store the value so it can be retrieved without fetching from the server again
     T value_;
 
-    // Store the min/max
-    // Its up to the user how they want to interpret min/max for types other than
-    // int/double
-    std::optional<T> min_;
-    std::optional<T> max_;
-
-    // Store the options of this parameter
-    std::vector<T> options_;
-
     // Store the name of the parameter
     std::string name_;
 
     // A list of functions to call when a new parameter value is set
-    std::vector<std::function<void(T)>> callback_functions;
+    // marked as mutable to allow const parameters to also register callbacks
+    mutable std::vector<std::function<void(T)>> callback_functions;
```

Comment:
very cool, TIL

Reply:
yeah its pretty sick, you have to mark` mutexes` as `mutable` if u want to use them in a const function